pkg/util/workerqueue/workerqueue.go

```diff
@@ -40,6 +45,10 @@ type WorkerQueue struct {
 	queue  workqueue.RateLimitingInterface
 	// SyncHandler is exported to make testing easier (hack)
 	SyncHandler Handler
+
+	mu      sync.Mutex
```

Comment:
Should this be a RWLock? I suppose there will only be a single thread hitting the `Healthy()` function, so there isn't much benefit to allow for multiple concurrent `Healthy()` requests. WDYT?

Reply:
I'm tempted to start with a plain mutex and leave open the possibility for improving performance of the `WorkerQueue` once we understand the bottle necks.

Reply:
SGTM!